yt/visualization/volume_rendering/off_axis_projection.py

```diff
@@ -444,21 +444,23 @@ def temp_weightfield(field, data):
         # We need the width of the plot window in projected coordinates,
         # i.e. we ignore the z-component
         wmax = width[:2].max()
-
-        # Normalize the positions & dx so that they are in the range [-0.5, 0.5]
-        xyz = np.stack(
-            [
-                ((data_source["index", k] - center[i]) / wmax).to("1").d
-                for i, k in enumerate("xyz")
-            ],
-            axis=-1,
+        # Recenter positions w.r.t. center of the plot window
+        xyz = data_source.ds.arr(
+            np.zeros((len(data_source[vol.field]), 3)), "code_length"
         )
 
+        # If we have periodic boundaries, we need to wrap the positions
         for idim, periodic in enumerate(data_source.ds.periodicity):
+            axis = "xyz"[idim]
```

Comment:
Do you need to be handling axis-ordering here? I can't remember if octree datasets are required to be ordered xyz.  e.g., 

```suggestion
            axis = data_source.ds.coordinates.axis_order[idim]
```